[tbb] fix shared tbb library configuration build under *nix

xandox opened the pull request:
**Describe the pull request**

- What does your PR fix? 
fix tbb cmake config for *nix systems if tbb was built not as static library

- Which triplets are supported/not supported? 
same as previous 
- Have you updated the CI baseline? 
not

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
I hope

Commit:
fix shared tbb library configuration build under *nix

Commit:
bump Port Version

JackBoosY committed:
[libmysql/libmariadb] Export unofficial cmake targets (#14290)

* [libmysql/libmariadb] Add vcpkg-cmake-wrapper and usage

* [libmysql] Export target unofficial-libmysql

* [libmariadb] Export target unofficial-libmariadb

* Add conditions

* Remove empty folder

* [libmysql] Enable static build, generate library mysqlclient

* Update ports/libmysql/CONTROL

* [libmysql] Re-write portfile.cmake

* [vcpkg_clean_executables_in_bin] clean up the pdbs

* [libmysql] Fix tools on non-Windows

* [libmysql] Fix tools, fix wrapper

* [libmysql] Add macro libmysql_FOUND

* [vcpkg_clean_executables_in_bin] Revert changes

* [libmysql] Export unofficial-libmysql on both dynamic and static

* Update ports/libmariadb/export-cmake-targets.patch

Co-authored-by: Billy Robert O'Neal III <bion@microsoft.com>
Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

cbergemann committed:
[libflac] require NASM for libflac[asm] (#14465)

* require NASM for libflac[asm]

* Increment port version number

* Update ports/libflac/portfile.cmake

Co-authored-by: BergemannC <CBergemann@nordex-online.com>
Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

NancyLi1013 committed:
[netcdf-cxx4] Export cmake targets (#14405)

* [netcdf-cxx4] Update to latest commit and also export targets file

* Revert the update to latest commit and make a patch to export cmake targets

* Revert the changes in fix-dependecy-hdf5.patch

Neumann-A committed:
[vcpkg/scripts] Add a way to get cmake compiler settings/flags (#12936)

* add function get_cmake_vars

* fine tuning.

* apply to make based ports.

* add log suffix on not windows platforms

* fix c&p error

* add previous LINK env

* setup env on windows and extract cpp flags correctly.

* Apply suggestions from code review

* commit changes from fontconfig PR

* Apply suggestions from code review

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* cleanup docs

* add conversion from somelib.lib to -lsomelib

* add missing ar-lib wrapper

* small but important regex correction

* add latest changes from update_fontconfig PR

* Apply suggestions from code review

first set which don't need special attention

Co-authored-by: ras0219 <533828+ras0219@users.noreply.github.com>

* Apply suggestions from code review

one more simple change

Co-authored-by: ras0219 <533828+ras0219@users.noreply.github.com>

* [x264] set env AS

* fix bugs due to refactor

* use subpath everywhere

* apply changes from CR

* remove unnecessary lines 41 & 44

* remove flag transformation

* reintroduce the flag / to - transformation for MSVC

* Apply suggestions from code review

Co-authored-by: ras0219 <533828+ras0219@users.noreply.github.com>

* rename the function

* rename function/variables

* transform flags list

* Apply suggestions from code review

* fix vcpkg_build_make due to the variable name change

* fix another case of function renaming regressions

* only rename config.log if it exists

* actually add the script after vcpkg_common_functions was deleted.

* remove setting of ldflags if path contains spaces

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>
Co-authored-by: ras0219 <533828+ras0219@users.noreply.github.com>

JackBoosY commented on ports/tbb/portfile.cmake:
```diff
@@ -24,7 +24,17 @@ if (NOT VCPKG_TARGET_IS_WINDOWS)
 
     # Settings for TBBConfigInternal.cmake.in
-    set(TBB_LIB_EXT a)
+    if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
+        set(TBB_LIB_EXT a)
+    else()
+        if (VCPKG_TARGET_IS_LINUX)
+            set(TBB_LIB_EXT "so.2")
+        elseif(VCPKG_TARGET_IS_OSX)
+            set(TBB_LIB_EXT "dylib")
+        else()
+            set(TBB_LIB_EXT "so")
+        endif()
+    endif()
```
Does it works? `TBB_LIB_EXT` is only used in `vcpkg_install_msbuild` which can't be used in *inx.

xandox replied:
>Does it work?

Yes :)
>TBB_LIB_EXT is only used in vcpkg_install_msbuild which can't be used in *inx.

Not, It also used in ${SOURCE_PATH}/cmake/templates/TBBConfigInternal.cmake.in during file configuration (Line - 105)

Commit:
installing missed TBBConfigVersion.cmake

NancyLi1013 committed:
[openssl] Remove unused patches (#14477)

* [openssl] Remove unused patches

* Update ports/openssl/CONTROL

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

OgreTransporter committed:
[libosmscout] Add new port (#14485)

* [libosmscout] Add new port

* [libosmscout] Update manifest format

* Update ports/libosmscout/portfile.cmake

Co-authored-by: Phoebe <20694052+PhoebeHui@users.noreply.github.com>

* Qt feature removed due to problems with various configurations.

* Change of line ends from Windows to Linux

* Disable ARM, UWP and static

* Update support tag

Co-authored-by: Phoebe <20694052+PhoebeHui@users.noreply.github.com>

Kwizatz committed:
[v8] Update v8 to current stable version: 8.6.395.17 (#13355)

ianormy committed:
[onnxruntime-gpu] Add port for onnxruntime (GPU) (#14481)

* feat: Add the port for onnxruntime-gpu
This port uses the binaries created by Microsoft for the onnxruntime
which will run with a GPU.

* Update ports/onnxruntime-gpu/portfile.cmake

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

Co-authored-by: Ian Ormesher <ian.ormesher@zeiss.com>
Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

LilyWangL committed:
[paho-mqtt] Update to version 1.3.5 (#13680)

* [paho-mqtt] Update to version 1.3.5

* [paho-mqtt] Fix build error

* [paho-mqttpp3] Fix build error

* [paho-mqttpp3] Fix build error

* [paho-mqttpp3] Fix build error

* [paho-mqtt] Fix old patch

* [paho-mqtt] Add vcpkg_copy_tools

* [paho-mqtt] Fix build error

NancyLi1013 committed:
[netcdf-c] Fix linkage error (#14421)

SeekingMeaning committed:
[xsimd][xtensor][xtl] Update to latest (#14454)

* [xsimd] update to 7.4.9

* [xtensor] update to 0.21.9

* [xtl] update to 0.21.9

SeekingMeaning committed:
[xeus] update to 0.24.3 (#14455)

JackBoosY committed:
[libxslt] Fix install tools and pkgconfig (#14476)

* [libxslt] Fix install tools and pkgconfig

* [libxml2] install pkgconfig

jube committed:
update to 0.18.1 (#14482)

BillyONeal committed:
Baseline fixes for 2020-11-07 (#14493)

redist-plus-plus: Remove some "passing, remove from fail lists"

ignition-msgs*: Nicole is working on fixing these here: https://github.com/microsoft/vcpkg/pull/14399

eao197 committed:
RESTinio updated to v.0.6.12. (#14496)

FabienPean committed:
[libigl] Fix dynamic build and enable header-only mode (#14376)

* [libigl] Fix dynamic build and enable header-only mode

* Update portfile.cmake

Co-authored-by: Lily <47812810+LilyWangL@users.noreply.github.com>

JackBoosY commented:
I will test the usage manually later.

jgehw committed:
[tensorflow-cc] More or less completely rewritten port, update to 2.3.1 (#13028)

* fix build issues occurring from default bazel config being used because customized bazel config is stored in wrong directory

* [tensorflow-cc] Update CONTROL and ci.baseline.txt

* fix also applies to windows static build

* fix debug and static builds as well as library naming for non-linux
- add patch to fix debug builds
- add patch to fix exports for static linking
- really build debug (instead of cloning release)
- override bazel build options for debug (work around bazel bug)
- bazel doesn't support static libraries: work around by building dynamic library and constructing static linkage commands from build log
- Windows .pdb file can't be >4GB even on x64: work around using reduced debug information
- Windows doesn't support .lib files >4GB even on x64, so split into multiple libs
- vcpkg requires equal amount of libs for debug and release: work around using handcrafted empty dummy libs
- fix naming of libs (.dll on Windows and .dylib on macOS)

* upgrade tensorflow from v1.14 to v2.3
- adapt patch files to tensorflow code changes
- update bazel from v0.25.2 to v3.1
- on Windows use python installed on the host instead of embedded python obtained via vcpkg because embedded python lacks pip, which we need to obtain numpy
- on Windows add MSYS2 to the PATH so that bazel tools can access MSYS2 GIT
- add support for custom CA certificates when using HTTPS_PROXY

* fix execute process macro
The existing implementation totally screwed up commands if the command's arguments contained semicolons (this is the case, e.g., in the FindPython modules of the cmake distribution).

* extend overriden execute_process to more than one COMMAND as there actually are use cases for this

* added another patch required for tensorflow v2.3, fixed path and working directory

* Revert "incorporate changes from microsoft:master"

* Revert "Revert "incorporate changes from microsoft:master""

* final fixes for static build + improving out messages

* enabling linux and osx in CI to see if it works now

* simplified code, fixed version numbers, fixed generated include cmake file

* fix failing postbuild check on handcrafted empty dummy library by spreading the last real libraries contents over the required number of libraries

* remove dead code commit by mistake again

* improvements from code review

* cleaner fix for debug code

* find pip3 in PATH (PYTHON3_DIR apparently not valid for pip3)

* fix error in python helper script

* fix wrong libname in postbuild script

* fix python detection + switch to python on msys2 (instead of embedded python) for Windows as we need numpy

* fix order of arguments

* fix command (it may contain spaces such as C:\Program Files\...)

* revert last commit (root cause for CI failures is something different: there are line breaks in path)

* fix regex comparision
(value needs to be escaped as it may contains regex special characters such as brackets, eg C:/Program Files (x86)/...)

* fix linebreaks in generated file

* fix CRT linkage
(macOS doesn't support static CRT linkage; it's set to dynamic even static target triplets for macOS and linux)

* refactor implemenation to avoid as much code duplication as possible -- algorithmically identical

* fix version numbers in helper scripts

* enable work-around for Windows until bazel fix is available

* install missing python3-pip on linux

* fix linux build by patching

* apply timeout feature now available via merged master branch

* correct linux build patch

* improve debug build patches
(no functional difference because LOG(FATAL, ...) macro internally anyway calls abort(), which the compiler doesn't detect in debug mode...

* improve linux patch

* temporarily add debug to inspect what's going on on macOS CI

* remove temporary debug code and fix static linking scripts for linux and macOS

* fix regex escaping

* fix ambiguous match while grepping for the framework link command

* extend fix of ambiguous match while grepping for the framework link command

* fix what merge of master broke

* fix more what got broken by merging master
(all packages and their dependencies are now maintained manually instead of using pacman...)

* remove "unofficial" from filename

* added switch do distinct classic and manifest mode when generating config.cmake file

* create symlinks for libraries without version number

* fix linux postbuild script

* temporarily disable code making problems

* add note for linking on Linux and macOS

* forget to add README file in previous commit

* add file forgotton in macro fixing patch

* fix python library path

* fix macOS static link command

* update linkage instructions in README

* Update ports/tensorflow-cc/CONTROL

* Update ports/tensorflow-cc/portfile.cmake

* Update scripts/ci.baseline.txt

* use vcpkg_execute_required_process

* pass C_FLAGS and CXX_FLAGS to bazel

* fix INTERFACE_INCLUDE_DIRECTORIES

* fix optional c/cxx arguments

* also add linker opts

* update README

* merge static libs into one
to support force_load (cannot force_load both due to duplicate symbols)

* update README

* quote python path (it might contain spaces that don't get escaped inside outer quotes of bash command)

* fix python path also for static build

* add arm(64) as currently unsupported arch

* bazel 3.7 is available -> remove workaround

* update README, remove necessary c-ares from deps

* update msys package

* add uwp specific options, and minor general improvements

* fix string replace

* fix control file and windows path separator

* revert backslashes-fix -- the root cause was missing .exe extension

* upgrade to tf 2.3.1

* fix hard-coded version

* remove uwp work-in-progress code so that PR can be merged

* update README and print out usage info in portfile

* Update ports/tensorflow-cc/README-linux

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/README-linux

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/README-linux

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/README-linux

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/README-linux

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/README-windows

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/portfile.cmake

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/portfile.cmake

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/portfile.cmake

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/portfile.cmake

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/portfile.cmake

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* improve usage hints as discussed in review

* add comment

* apply changes from review

* make additional compiler / linker args space-proof

* Update ports/tensorflow-cc/README-macos

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/tensorflow-cc-config-shared.cmake.in

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/tensorflow-cc-config-shared.cmake.in

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/README-linux

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update ports/tensorflow-cc/README-macos

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* apply changes from code review

* maybe fix the config files

* rob.maynard CRs

* fix windows static lib naming for first part

* Update ports/tensorflow-cc/generate_static_link_cmd_windows.py

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* apply rob maynards 1st suggestion

* nicole CRs

* format/fix-compile

* fix missing string termination

* prefer IMPORTED_LOCATION over IMPORTED_LOCATION_RELEASE to have default fall-back

* hopefully fix the issue where no libraries are generated

* final stuff

Co-authored-by: Gehweiler <Joachim_Gehweiler@McAfee.com>
Co-authored-by: wangli28 <wangli28@beyondsoft.com>
Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>
Co-authored-by: Joachim Gehweiler <joachim@Joachims-iMac.local>
Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

strega-nil committed:
[vcpkg] Add x-ignore-lock-failures (#14397)

* [vcpkg] Add disable-lock options

This should fix #14281 at least as a stopgap.

Additionally, adds better errors.

* billy CRs

* change from "disable-lock" to "allow-spurious-lock-failures"

* billy cr

JackBoosY committed:
[libmariadb] Also export target mariadbclient (#14533)

Neumann-A committed:
[harfbuzz] fix icu linkage (#14530)

JackBoosY committed:
[curl] Enable feature tool and install curl-config on non-Windows (#14473)

* [curl] Enable feature tool and install curl-config on non-Windows

* Update ports/curl/portfile.cmake

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

ousttrue committed:
[usd] fix dll path and update to 20.08 (#13687)

* fix plugInfo.json and dll path

* release build issue: https://github.com/PixarAnimationStudios/USD/issues/960

* add usage

* [usd] Add homepage

* [usd] Fix install copyright

* fix for linux build

* fix windows build

* fix No such file or directory

* use VCPKG_TARGET_IS_WINDOWS

* cmake to share/pxr

* fix TARGET_PATH

Co-authored-by: Lily <47812810+LilyWangL@users.noreply.github.com>
Co-authored-by: wangli28 <wangli28@beyondsoft.com>
Co-authored-by: Nicole Mazzuca <mazzucan@outlook.com>

dimztimz committed:
[nuspell] Update version to 4.0.1 (#14505)

* [nuspell] Update version to 4.0.1

* [nuspell] Install pkg-config files instead of deleting them.

JackBoosY committed:
[ampl-mp] Update version, separate port ampl-asl and fix arm build (#14518)

* [ampl-asl] Separate port ampl-mp module asl to ampl-asl

* [ampl-mp] Update version, separate port ampl-asl and fix arm build

* update baseline

* [VMSS] Pre-install ampl-mp:x86-windows

* Update ports/ampl-asl/portfile.cmake

mathisloge committed:
[comms ] update to the next version (#14521)

* update comms to the next version; fixed github repo

* Update portfile.cmake

added the no warn as error again

* use only necessary boost packages as dependencies

* fix formatting

Commit:
[realsense2] Update to version 2.39.0 (#14522)

mathisloge committed:
[comms-ublox] added new port (#14523)

* add comms-ublox

* forgot the dependencies

* Fix naming of cmake variables

JackBoosY committed:
[freetype] Fix INTERFACE_LINK_LIBRARIES in exported cmake file (#14499)

* [freetype] Fix INTERFACE_LINK_LIBRARIES in exported cmake file

* [freetype] Bump version

* [libpng] Fix dependency zlib in INTERFACE_LINK_LIBRARIES

* [freetype] Revert changes about keyword PUBLIC and pkgconfig

Neumann-A committed:
[ZSTD] Update to 1.4.5 (#14395)

* [ZSTD] Update to 1.4.5

* [zstd] fix wrong assert to fix uwp regressions

* always install *.pc file. Correct -l flag in pc file

* fix orc regression

* fix arrow regressions

* fix orc zstd linkage

* fix find_package(ZSTD)

* install zstd dlls.

* fix the patch due to a replacement error

Co-authored-by: Billy Robert O'Neal III <bion@microsoft.com>

AkiaCode committed:
[vcpkg] Add korean version Readme (#14310)

* Project Start

Co-authored-by:  jiyuPk
<wltmfdmsdnjf@gmail.com>

* Update README_ko_KR.md

1차 문서 거의 다 완성 중

Co-Authored-By: GU <64261175+jiyuPk@users.noreply.github.com>

* Update README_ko_KR.md

* Fix typo ko_KR

Co-authored-by: GU <64261175+jiyuPk@users.noreply.github.com>
Co-authored-by: Lily <47812810+LilyWangL@users.noreply.github.com>

strega-nil committed:
[3fd] Update to 2.6.3, fix port (#14528)

* [3fd] Update to 2.6.3, fix port

* update baseline

* JackBoosY CRs

Co-authored-by: JackBoosY <yuzaiyang@beyondsoft.com>

strega-nil committed:
[llvm] install tools in tools/llvm (#14399)

* [llvm] install tools in tools/llvm

* remove the giant commented out block of code

* update port-version

* hopefully fix the port?

There are still some issues, namely that the python scripts and DLLs that are copied into `tools/llvm`
are not considered to be "installed by llvm", and thus are not removed when llvm is removed

* format!

* apparently REGEX REPLACE fails if a thing doesn't match

* fix LLVM_REMOVE_EXTENSION_REGEX on windows

* actually read the cmake regex docs...

* fix the name of the variable

* turns out CMAKE_MATCH_1 is the one I want

* need to update VERSION for new policy

* stop removing debug/bin

* fix faulty merge

Commit:
Merge branch 'master' of github.com:xandox/vcpkg

JackBoosY commented:
Did you pushed the wrong branch?

xandox commented:
Hell!
Yes. Sorry. I will recreate PR.

xandox commented:
Created new https://github.com/microsoft/vcpkg/pull/14555
Sorry again. My git-fu not perfect.
